gateway-ha/src/main/java/io/trino/gateway/ha/resource/GatewayHealthCheckResource.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.resource;
+
+import com.google.inject.Inject;
+import io.trino.gateway.ha.clustermonitor.ActiveClusterMonitor;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Response;
+
+import static java.util.Objects.requireNonNull;
+
+@Path("/trino-gateway")
+public class GatewayHealthCheckResource
+{
+    private final ActiveClusterMonitor activeClusterMonitor;
+
+    @Inject
+    public GatewayHealthCheckResource(ActiveClusterMonitor activeClusterMonitor)
+    {
+        this.activeClusterMonitor = requireNonNull(activeClusterMonitor, "activeClusterMonitor is null");
+    }
+
+    @GET
+    @Path("/livez")
+    public Response liveness()
+    {
+        return Response.ok("ok").build();
```

Comment:
Do you see any downside to verify if any backend trino services are configured? not the status of them but just if any trino service exists?

Reply:
Yeah .. that might make sense .. a liveness of the app with no configured backends is literally not usable .. at the same time .. are there cases where this might be valid anyway .. say you start it up.. its live and then you use the UI to configure backends?